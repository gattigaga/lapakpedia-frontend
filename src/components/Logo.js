import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Text = styled.h1`
  font-size: ${props => props.fontSize}px;
  font-family: Roboto;
  margin: 0px;
`;

const Logo = props => <Text {...props}>Lapakpedia</Text>;

Logo.propTypes = {
  fontSize: PropTypes.number,
  className: PropTypes.string
};

Logo.defaultProps = {
  fontSize: 32
};

export default Logo;
