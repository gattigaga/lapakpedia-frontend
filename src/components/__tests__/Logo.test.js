import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Logo from "components/Logo";

describe("Logo", () => {
  const setup = propOverrides => {
    const props = {
      ...propOverrides
    };

    const wrapper = shallow(<Logo {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom font size", () => {
    const { wrapper } = setup({ fontSize: 48 });

    expect(wrapper.props().fontSize).toEqual(48);
  });
});
