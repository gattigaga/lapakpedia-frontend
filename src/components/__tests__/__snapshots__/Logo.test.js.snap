// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should renders default 1`] = `
<styled.h1
  fontSize={32}
>
  Lapakpedia
</styled.h1>
`;
